quickwit-core/src/index.rs

```diff
@@ -0,0 +1,83 @@
+/*
+    Quickwit
+    Copyright (C) 2021 Quickwit Inc.
+
+    Quickwit is offered under the AGPL v3.0 and as commercial software.
+    For commercial licensing, contact us at hello@quickwit.io.
+
+    AGPL:
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use quickwit_doc_mapping::DocMappingMetadata;
+use quickwit_metastore::{MetastoreUriResolver, SplitState};
+
+type IndexUri = String;
+
+// anyhow errors are fine for now but we'll want to move to a proper error type eventually.
+pub async fn create_index(
```

Comment:
I think we can put those in different files eventually. No need to do that in that PR though.